packages/core/src/factory.ts

```diff
@@ -81,14 +80,22 @@ export function factory<Resource, Input, Output, Config, LoadOpts>(
         let output: Output | undefined;
         if (resource && components.mocker && (configObj as IPrismConfig).mock) {
           // generate the response
-          output = components.mocker.mock(
-            {
-              resource,
-              input: { validations: { input: inputValidations }, data: input },
-              config: configObj,
-            },
-            defaultComponents.mocker,
-          );
+          output = components.mocker
+            .mock(
+              {
+                resource,
+                input: { validations: { input: inputValidations }, data: input },
+                config: configObj,
+              },
+              defaultComponents.mocker,
+            )
+            .run(components.logger.child({ name: 'MOCKER', input }))
+            .fold(
+              e => {
+                throw e;
+              },
+              r => r,
```

Comment:
`r`?  how about `x`? :trollface: 
Does it stand for `result`?

Reply:
Yes, it's a shortcut for result, but I can change it, I do not mind

Reply:
This clearly is `identity`. Can use `_.identity`

Reply:
@lag-of-death I can make the change, but is it going to improve the readability? At first glance it makes me scratch a little bit more my head instead of clarifying. What do you think?

Reply:
@XVincentX, honestly I don't care, `x => x`, `r => r`, `_.identity` are all the same to me. The fact is though, that `x => x` and `r => r` might be more understandable for those, that don't know about `identity`. Up to you, if it was up to me, I'd most probably left it as is ;)